c-type-name.tex

```diff
@@ -0,0 +1,50 @@
+\begin{syntax}
+C-type-expr ::= specifier-qualifier+ abstract-declarator?
+\
+C-type-name ::= declaration-specifier+
+\
+specifier-qualifier ::= type-specifier | type-qualifier
+\
+type-qualifier ::= "const" | "volatile"
+\
+type-specifier ::= "void" ;
+                   | "char" ;
+                   | "short" ;
+                   | "int" ;
+                   | "long" ;
+                   | "float" ;
+                   | "double" ;
+                   | "signed" ;
+                   | "unsigned" ;
+                   | ( "struct" | "union" | "enum" ) ident [ACSL does not permit declaring a new type within the type-specifier] ;
+                   | <enum-specifier> ;
+                   | ident
+\
+abstract-declarator ::= pointer ;
+| pointer direct-abstract-declarator ;
+| direct-abstract-declarator
+\
+pointer ::= ( "*" type-qualifier* )+
+\
+direct-abstract-declarator ::= "(" abstract-declarator ")" ; 
+| direct-abstract-declarator? "[" constant-expression "]" ;
```

Comment:
I'd make the `constant-expression` optional (this is explicitly mentioned in section 2.2.7).